src/components/product/image-preview-card.svelte

```diff
@@ -0,0 +1,79 @@
+<script>
+  import { createEventDispatcher } from 'svelte';
+  import Card from 'ui/card.svelte';
+  import Button from 'ui/button.svelte';
+  import getBackground from '@/utils/optimal-color.js';
+
+  export let image;
+  export let color;
+  export let index;
+  export let url = null;
+  let dataURL;
+  let loading = true;
+  let error = false;
+
+  if (url == null) {
+    let fileReader = new FileReader();
+    fileReader.readAsDataURL(image.file);
+    fileReader.onloadend = () => {
+      dataURL = fileReader.result;
+    };
+
+    image.promise.then(resp => {
+      loading = false;
+      if (!resp.ok) {
+        error = true;
+      }
+    });
+  } else {
+    loading = false;
+  }
+
+  const dispatch = createEventDispatcher();
+</script>
+
+<Card classname="image with-image {loading ? 'loading': ''} {error ? 'error' : ''}">
+  <div class="image-wrapper">
+    <img
+      src={url || dataURL || '/images/create-product/placeholder.svg'}
+      style="background: {getBackground(color)}"
+      alt={url || image.file.name}
+    />
+    <div class="status-overlay">
+      <div class="lds-ellipsis">
+        <div></div><div></div><div></div><div></div>
+      </div>
+    </div>
+  </div>
+  {#if url}
+    <div class="actions">
+      <svg src="images/icons/drag-handle.svg" class="drag-handle" />
+      <Button
+        isDanger
+        isRound
+        tooltip="Delete image"
+        on:click={(e) => { e.preventDefault(); dispatch('delete-file', { fileIndex: index }); }}
```

Comment:
I think the Svelte way would be like:
```suggestion
        on:click|preventDefault={(e) => { dispatch('delete-file', { fileIndex: index }); }}
```
though I'm not sure if it will work on elements that are not native HTML

Reply:
Yep. it won't. Only `once` can be applied to components

Reply:
So it is about the component and not the event being a DOM event